crates/factor-outbound-pg/tests/factor_test.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::bail;
+use spin_factor_outbound_networking::OutboundNetworkingFactor;
+use spin_factor_outbound_pg::OutboundPgFactor;
+use spin_factor_variables::{StaticVariables, VariablesFactor};
+use spin_factor_wasi::{DummyFilesMounter, WasiFactor};
+use spin_factors::{anyhow, RuntimeFactors};
+use spin_factors_test::{toml, TestEnvironment};
+use spin_world::v2::postgres::HostConnection;
+
+#[derive(RuntimeFactors)]
+struct TestFactors {
+    wasi: WasiFactor,
+    variables: VariablesFactor,
```

Comment:
Why do we include this here? Does `OutboundNetworkingFactor` unconditionally depend on `VariablesFactor`? I would have assumed that it would only depend on it if a variable was used but perhaps not?

Reply:
Yeah, the dependency is unconditional. Making it optional would require some refactoring in `AllowedHostsConfig` to detect whether `allowed_outbound_hosts` needs variables, and more generally might make factor dependencies more difficult to reason about.